### PR TITLE
Migrate net5

### PR DESCRIPTION
**Pull request by zhuliice**
.Net 5 has been released for a while, this PR is migrating forge-api-dotnet-core to `.Net 5` from `netstandard2.0`, so we have better support from latest libraries

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+### 2.0.0.0
+
+* Migrate to .Net 5
+* Support to return HttpStatusCode in HttpRequestException
+
```
Comment (zhuliice) [inline CHANGELOG.md:5]: update CHANGELOG

File: README.md [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
 
-- .NET Standard 2.0 or later
+- .NET 5 or later
 
 ### Dependencies
 
-- [Newtonsoft.Json](https://github.com/JamesNK/Newtonsoft.Json) 12.0.1 or later
-- [Polly](https://github.com/App-vNext/Polly) 7.2.1 or later
+- [Newtonsoft.Json](https://github.com/JamesNK/Newtonsoft.Json) 13.0.1 or later
+- [Polly](https://github.com/App-vNext/Polly) 7.2.2 or later
```
Comment (zhuliice) [inline README.md:17]: update README.md about project's information

File: src/Autodesk.Forge.Core.E2eTestHelpers/Autodesk.Forge.Core.E2eTestHelpers.csproj [hunk 3]
```diff
@@ -1,20 +1,24 @@
...
-    <TargetFrameworks>netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>net5.0</TargetFrameworks>
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
     <Authors>Autodesk Forge Team</Authors>
     <Company>Autodesk</Company>
     <Product>Autodesk Forge</Product>
     <Description>Shared code for Forge client sdks e2e tests</Description>
     <Copyright>Autodesk Inc.</Copyright>
-    <Version>1.0.3</Version>
+    <Version>2.0.0</Version>
```
Comment (zhuliice) [inline src/Autodesk.Forge.Core.E2eTestHelpers/Autodesk.Forge.Core.E2eTestHelpers.csproj:11]: bump version

**Review by szilvaa — APPROVED** [review]

File: src/Autodesk.Forge.Core.E2eTestHelpers/Autodesk.Forge.Core.E2eTestHelpers.csproj [hunk 4]
```diff
@@ -1,20 +1,24 @@
...
     <PackageProjectUrl>https://github.com/Autodesk-Forge/forge-api-dotnet-core</PackageProjectUrl>
-    <PackageIconUrl>http://developer.static.autodesk.com/images/logo_forge-2-line.png</PackageIconUrl>
+    <PackageIcon>logo_forge-2-line.png</PackageIcon>
     <PackageReleaseNotes>For full release notes see https://github.com/Autodesk-Forge/forge-api-dotnet-core/blob/master/CHANGELOG.md</PackageReleaseNotes>
   </PropertyGroup>
 
+  <ItemGroup>
+    <None Include="..\..\images\logo_forge-2-line.png" Pack="true" PackagePath="\"/>
+  </ItemGroup>
+
```
Comment (zhuliice) [inline src/Autodesk.Forge.Core.E2eTestHelpers/Autodesk.Forge.Core.E2eTestHelpers.csproj:21]: try to get rid of warning: https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu5048
>The 'PackageIconUrl'/'iconUrl' element is deprecated. Consider using the 'PackageIcon'/'icon' element instead. Learn more at https://aka.ms/deprecateIconUrl

File: src/Autodesk.Forge.Core/ForgeHandler.cs [hunk 5]
```diff
@@ -63,9 +63,9 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
             IAsyncPolicy<HttpResponseMessage> policies;
 
             // check if request wants custom timeout
-            if (request.Properties.TryGetValue(ForgeConfiguration.TimeoutKey, out var timeoutValue))
```
Comment (zhuliice) [inline src/Autodesk.Forge.Core/ForgeHandler.cs:66]: `HttpRequestMessage.Properties` is deprecated, use `Options` instead

File: src/Autodesk.Forge.Core/HttpResponseMessageExtensions.cs [hunk 6]
```diff
@@ -38,7 +38,7 @@ public static async Task<HttpResponseMessage> EnsureSuccessStatusCodeAsync(this
                 {
                     errorMessage = $"\nMore error details:\n{errorMessage}.";
                 }
-                throw new HttpRequestException($"The server returned the non-success status code {(int)msg.StatusCode} ({msg.ReasonPhrase}).{errorMessage}");
+                throw new HttpRequestException($"The server returned the non-success status code {(int)msg.StatusCode} ({msg.ReasonPhrase}).{errorMessage}", null, msg.StatusCode);
             }
```
Comment (zhuliice) [inline src/Autodesk.Forge.Core/HttpResponseMessageExtensions.cs:42]: fix the issue reported here: https://github.com/Autodesk-Forge/forge-api-dotnet-core/issues/70#issue-890954746

Comment (zhuliice) [reply src/Autodesk.Forge.Core/HttpResponseMessageExtensions.cs:42]: @szilvaa-adsk there is another requirement: https://github.com/Autodesk-Forge/forge-api-dotnet-core/issues/70#issuecomment-841099809
Should we fix it? and How to do it?

Comment (szilvaa) [reply src/Autodesk.Forge.Core/HttpResponseMessageExtensions.cs:42]: I think we should derive a new `TooManyRequestsException` from `HttpRequestException` and throw that specifically. This new exception could then have a `TimeSpan RetryAfter` property with the value from the header.

**Pull request merged**